### PR TITLE
Alot of updates

### PR DESCRIPTION
**Pull request by CadetMercury615**
- Created ServerOptions
- Made Server Class implement ServerOptions
- Created Server Info command using info from ServerOptions interface

### CODE REVIEW

File: src/command/default/ServerInfo.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        this.restricted = true;
+    }
+
+    public onRun(sender: CommandSender, label: string, args: Array<string>): void {
+        sender.sendMessage(`Server Name: ${this.getServer().ServerName}`)
+        sender.sendMessage(`Server Address: ${this.getServer().Server_IP}`)
+        sender.sendMessage(`Server Port: ${this.getServer().Port}`)
+        sender.sendMessage(`Server MOTD: ${this.getServer().MOTD}`)
+        sender.sendMessage(`Max Players: ${this.getServer().MaxPlayers}`)
+
```
Comment (john-bv) [inline src/command/default/ServerInfo.ts:33]: Syntax here can be improved

File: src/command/default/ServerInfo.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+class ServerInfoCommand extends Command {
+    public prepare(): void {
+        this.permissions = [];
+        this.permissions.push('netrex.command.stop');
+        this.operators = true;
+        this.restricted = true;
+    }
+
+    public onRun(sender: CommandSender, label: string, args: Array<string>): void {
+        sender.sendMessage(`Server Name: ${this.getServer().ServerName}`)
```
Comment (john-bv) [inline src/command/default/ServerInfo.ts:28]: Keep same codestyle

File: src/command/default/ServerInfo.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+class ServerInfoCommand extends Command {
+    public prepare(): void {
+        this.permissions = [];
+        this.permissions.push('netrex.command.stop');
+        this.operators = true;
+        this.restricted = true;
+    }
+
+    public onRun(sender: CommandSender, label: string, args: Array<string>): void {
```
Comment (john-bv) [inline src/command/default/ServerInfo.ts:27]: Array<string> should be typed as `string[]`

File: src/server/Server.ts [hunk 4]
```diff
@@ -16,8 +16,23 @@ import Logger from '@/utils/Logger';
 import CommandManager from '@/command/CommandManager';
 import RakNet from '@/network/rakNet/RakNet';
 import PacketManager from '@/network/bedrock/PacketManager';
+import { ServerOptions } from './ServerOptions'
 
-class Server extends EventEmitter {
+class Server extends EventEmitter implements ServerOptions {
+    ServerName = "NetrexTestServer"
```
Comment (john-bv) [inline src/server/Server.ts:22]: No use of public or private deceleration, needs changing.
Issues:
- Declared as class constants with this case
- Should be initialized with passed options
- PVP will be handled by plugins
- RCON will be its separate plugin
- Hardcore, not sure what you mean by this

File: src/server/ServerOptions.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export interface ServerOptions {
+    ServerName: string;
+    Server_IP: string;
```
Comment (john-bv) [inline src/server/ServerOptions.ts:3]: Refrain from mixing naming conventions, use `CamelCase`not `under_score`

File: src/server/ServerOptions.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    Port: number;
+    MOTD: string;
+    MaxPlayers: number;
+    DefaultGamemode: number; // 0 for survival, 1 for creative
+    EnableQuery: boolean; //true for yes, false for no
+    EnableRCON: boolean; //true for yes, false for no
+    RCONPassword: string;
```
Comment (john-bv) [inline src/server/ServerOptions.ts:10]: Should be separate interface for rcon options

File: src/server/Server.ts [hunk 7]
```diff
@@ -16,8 +16,23 @@ import Logger from '@/utils/Logger';
...
+    Server_IP = "127.0.0.1" // If you are running the server on your local machine, please change the server ip to SERVER_IP
+    Port = 19132 // By Default this will be 19132
+    MaxPlayers = 5; // Only change this the recommended limit that your host suggests
+    MOTD = "Test Server : 1.14.6"
+    DefaultGamemode = 0; // 0 for survival, 1 for creative
+    EnableQuery = true; //true for yes, false for no
+    EnableRCON = true; //true for yes, false for no
+    RCONPassword = "Olybear9IsStinky";
```
Comment (john-bv) [inline src/server/Server.ts:30]: Please refrain from writing this in pull requests

File: src/command/default/ServerInfo.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+ */
+import Command from '@/command/Command';
+import CommandSender from '@/command/senders/CommandSender';
+import CommandManager from '../CommandManager';
+import Server from '@/server/Server';
+
+class ServerInfoCommand extends Command {
+    public prepare(): void {
+        this.permissions = [];
+        this.permissions.push('netrex.command.stop');
```
Comment (john-bv) [inline src/command/default/ServerInfo.ts:22]: No point in pushing when you can simply set the array here.

**Comment by Olybear9** [comment]
Server options should be dynamic per request of the server owner.

File: src/command/default/ServerInfo.ts (hunk 2 above)
Comment (CadetMercury615) [reply src/command/default/ServerInfo.ts:28]: k dad

File: src/command/default/ServerInfo.ts (hunk 1 above)
Comment (CadetMercury615) [reply src/command/default/ServerInfo.ts:33]: k sis

File: src/command/default/ServerInfo.ts (hunk 3 above)
Comment (CadetMercury615) [reply src/command/default/ServerInfo.ts:27]: k mom

File: src/server/Server.ts (hunk 4 above)
Comment (CadetMercury615) [reply src/server/Server.ts:22]: k bro

File: src/server/ServerOptions.ts (hunk 5 above)
Comment (CadetMercury615) [reply src/server/ServerOptions.ts:3]: k son

File: src/server/ServerOptions.ts (hunk 6 above)
Comment (CadetMercury615) [reply src/server/ServerOptions.ts:10]: k stepsis

File: src/server/Server.ts (hunk 7 above)
Comment (CadetMercury615) [reply src/server/Server.ts:30]: k

File: src/command/default/ServerInfo.ts (hunk 8 above)
Comment (CadetMercury615) [reply src/command/default/ServerInfo.ts:22]: k

File: src/command/default/ServerInfo.ts (hunk 1 above)
Comment (CadetMercury615) [reply src/command/default/ServerInfo.ts:33]: I'll use a function for this^

File: src/server/ServerOptions.ts (hunk 6 above)
Comment (CadetMercury615) [reply src/server/ServerOptions.ts:10]: I'll make separate file

**Comment by Bavfalcon9** [comment]
Closing as this is no longer relevant

**Pull request closed without merging**